react-components/src/components/RevealToolbar/RuleBasedOutputsButton.tsx

```diff
@@ -22,35 +22,57 @@ import { useTranslation } from '../i18n/I18n';
 import { useFetchRuleInstances } from '../RuleBasedOutputs/hooks/useFetchRuleInstances';
 import { type AssetStylingGroup } from '../..';
 
+type EmptyRule = {
+  rule: any;
+  isEnabled: boolean;
+};
 type RuleBasedOutputsButtonProps = {
   onRuleSetStylingChanged?: (stylings: AssetStylingGroup[] | undefined) => void;
+  onRuleSetSelectedChanged?: (ruleSet: RuleAndEnabled | EmptyRule | undefined) => void;
 };
 export const RuleBasedOutputsButton = ({
-  onRuleSetStylingChanged
+  onRuleSetStylingChanged,
+  onRuleSetSelectedChanged
 }: RuleBasedOutputsButtonProps): ReactElement => {
   const [currentRuleSetEnabled, setCurrentRuleSetEnabled] = useState<RuleAndEnabled>();
-  const [ruleInstances, setRuleInstances] = useState<RuleAndEnabled[]>();
+  const [emptyRuleSelected, setEmptyRuleSelected] = useState<EmptyRule>();
+  const [ruleInstances, setRuleInstances] =
+    useState<Array<RuleAndEnabled | EmptyRule | undefined>>();
   const { t } = useTranslation();
 
   const ruleInstancesResult = useFetchRuleInstances();
 
+  const emptySelection: EmptyRule = {
```

Comment:
This should be defined only inside the `onChange`-callback

Reply:
https://github.com/cognitedata/reveal/pull/4453/commits/5824d886e02f0d1ee6ecdf3c0738b52e885dd19f